spec/reek/ast/object_refs_spec.rb

```diff
@@ -2,57 +2,16 @@
 require_lib 'reek/ast/object_refs'
 
 RSpec.describe Reek::AST::ObjectRefs do
-  let(:refs) { Reek::AST::ObjectRefs.new }
+  let(:refs) { described_class.new }
 
   context 'when empty' do
-    it 'should report no refs to self' do
+    it 'reports no refs to self' do
       expect(refs.references_to(:self)).to be_empty
     end
   end
 
-  context 'with references to a, b, and a' do
-    context 'with no refs to self' do
-      before(:each) do
-        refs.record_reference(name: :a)
-        refs.record_reference(name: :b)
-        refs.record_reference(name: :a)
-      end
-
-      it 'should report no refs to self' do
-        expect(refs.references_to(:self)).to be_empty
-      end
-
-      it 'should report :a as the max' do
-        expect(refs.most_popular).to include(:a)
-      end
-
-      it 'should not report self as the max' do
-        expect(refs.self_is_max?).to eq(false)
-      end
-
-      context 'with one reference to self' do
-        before(:each) do
-          refs.record_reference(name: :self)
-        end
-
-        it 'should report 1 ref to self' do
-          expect(refs.references_to(:self).size).to eq(1)
-        end
-
-        it 'should not report self among the max' do
-          expect(refs.most_popular).to include(:a)
-          expect(refs.most_popular).not_to include(:self)
-        end
-
-        it 'should not report self as the max' do
-          expect(refs.self_is_max?).to eq(false)
-        end
-      end
-    end
-  end
```

Comment:
Yes, they were essentially a duplicate of the specs below, but with more nesting :smile:.